src/main/java/quest_system/QuestManager.java

```diff
@@ -36,10 +33,24 @@ public Quest getQuest(String name) {
     /**
      * Mark a quest_system.Quest as completed and issue rewards.
      * @param player the player Character completing the quest_system.Quest
-     * @param name the name of the quest_system.Quest to be completed
+     * @param quest the quest_system.Quest to be completed
      */
-    public void completeQuest(GameCharacter player, String name) {
-        c.complete(player, this.getQuest(name));
+    public void completeQuest(GameCharacter player, Quest quest) {
+        assert !quest.isComplete();
```

Comment:
are we catching the exception somewhere?

Reply:
nope